### ISSUE TITLE
[feature request] An option to obfuscate/hide remote IP addresses in the log

### ISSUE THREAD

**Issue by epoberezkin** [coturn/coturn#911]
As per title, for privacy

Thank you!

**Comment by ggarber** [comment]
No, that's not supported.  You could send the logs to stdout and pipe them to a script that filters the ip addresses.  That's the only idea I have that doesn't involve forking coturn right now.  Maybe that's good enough for you;

**Comment by jonesmz** [comment]
Some additional context for this would be the European Union's GDPR law that heavily discourages logging IP addresses.

**Comment by epoberezkin** [comment]
@jonesmz interesting - can you provide a reference to GDPR provision about it? I don't remember seeing it there.

**Comment by jonesmz** [comment]
IP addresses are considered personally identifiable information (PII), and storing logs of PII isn't allowed unless doing so is critical to the application.

I don't know the specific section of the GDPR, I'm instead operating based on the training that my work does for developers to ensure GDPR compliance. We went through our whole codebase over the last few years to purge all places where we logged PII, including IP addresses.

**Comment by epoberezkin** [comment]
> IP addresses are considered personally identifiable information (PII), and storing logs of PII isn't allowed unless doing so is critical to the application.

I believe you are confusing PII with online identifiers: https://gdpr.eu/recital-30-online-identifiers-for-profiling-and-identification/

There are some guidelines for their use, but it's not the same as PII.

**Comment by jonesmz** [comment]
I'm only repeating what my companies lawyers have presented to my development team.

I've always thought that pretending an ipaddress identified a person was a silly concept since back in the days of Napster, but here we are.

**Comment by eakraly** [comment]
Hi @epoberezkin 
Except for "we are open for contributions" I do not have a good answer.
I'll try to get a few hours of work into this feature  soon.

**Comment by epoberezkin** [comment]
@eakraly wasn't expecting you to do it :)

We don't need it right now strictly speaking.

IP addresses are needed for automatic firewall management, so you can add DDoS addresses to firewall rules.
